From 552d628c887d970b9a97d8db2629adc4820fb8e3 Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@freescale.com>
Date: Thu, 16 Jul 2015 14:44:24 +0530
Subject: [PATCH 194/226] staging: fsl-mc: export irq cleanup for vfio to use

VFIO driver needs these basic functions for
setting up itt/its of dprc's bound to it.

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@freescale.com>
(Stuart: resolved merge conflict, commit log cleanup)
Signed-off-by: Stuart Yoder <stuart.yoder@nxp.com>
---
 drivers/staging/fsl-mc/bus/dprc-driver.c    |    4 ++--
 drivers/staging/fsl-mc/include/mc-private.h |    4 ++++
 2 files changed, 6 insertions(+), 2 deletions(-)

--- a/drivers/staging/fsl-mc/bus/dprc-driver.c
+++ b/drivers/staging/fsl-mc/bus/dprc-driver.c
@@ -193,7 +193,7 @@ static void dprc_add_new_devices(struct
 	}
 }
 
-static void dprc_init_all_resource_pools(struct fsl_mc_device *mc_bus_dev)
+void dprc_init_all_resource_pools(struct fsl_mc_device *mc_bus_dev)
 {
 	int pool_type;
 	struct fsl_mc_bus *mc_bus = to_fsl_mc_bus(mc_bus_dev);
@@ -234,7 +234,7 @@ static void dprc_cleanup_resource_pool(s
 	WARN_ON(free_count != res_pool->free_count);
 }
 
-static void dprc_cleanup_all_resource_pools(struct fsl_mc_device *mc_bus_dev)
+void dprc_cleanup_all_resource_pools(struct fsl_mc_device *mc_bus_dev)
 {
 	int pool_type;
 
--- a/drivers/staging/fsl-mc/include/mc-private.h
+++ b/drivers/staging/fsl-mc/include/mc-private.h
@@ -157,4 +157,8 @@ int fsl_mc_populate_irq_pool(struct fsl_
 
 void fsl_mc_cleanup_irq_pool(struct fsl_mc_bus *mc_bus);
 
+void dprc_init_all_resource_pools(struct fsl_mc_device *mc_bus_dev);
+
+void dprc_cleanup_all_resource_pools(struct fsl_mc_device *mc_bus_dev);
+
 #endif /* _FSL_MC_PRIVATE_H_ */
